src/test/java/org/openjdk/shenandoah/CounterTest.java

```diff
@@ -0,0 +1,184 @@
+ *     2 along with this work; if not, write to the Free Software Foundation,
+ *     Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *     Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *     or visit www.oracle.com if you need additional information or have any
+ *     questions.
+ */
+package org.openjdk.shenandoah;
+
+import org.HdrHistogram.Histogram;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.openjdk.shenandoah.RegionAffiliation.OLD;
+import static org.openjdk.shenandoah.RegionAffiliation.YOUNG;
+import static org.openjdk.shenandoah.RegionState.*;
+
+public class CounterTest {
+    List<RegionStat> emptyUncommittedAge0 = new ArrayList<>();
+    List<RegionStat> emptyCommittedAge3 = new ArrayList<>();
+    List<RegionStat> trashAge6 = new ArrayList<>();
+    List<RegionStat> humongousAge9 = new ArrayList<>();
+    List<RegionStat> pinnedHumongousAge12 = new ArrayList<>();
+    List<RegionStat> cSetAge15 = new ArrayList<>();
+    List<RegionStat> pinned = new ArrayList<>();
+    List<RegionStat> pinnedCSet = new ArrayList<>();
+    List<RegionStat> tLab = new ArrayList<>();
+    List<RegionStat> gcLab = new ArrayList<>();
+    List<RegionStat> pLab = new ArrayList<>();
+    List<RegionStat> sharedLabYoung = new ArrayList<>();
+    List<RegionStat> sharedLabOld = new ArrayList<>();
+    Snapshot snapshot_1;
+    Snapshot snapshot_2;
+    Snapshot snapshot_3;
+    Snapshot snapshot_4;
+    Snapshot snapshot_5;
+    Snapshot snapshot_6;
+    Snapshot snapshot_7;
+    Snapshot snapshot_8;
+    Snapshot snapshot_9;
+    Snapshot snapshot_10;
+    Snapshot snapshot_11;
+    Snapshot snapshot_12;
+    Snapshot snapshot_13;
+    @Before
+    public void setup() {
+        for (int i = 0; i < 10; i++) {
+            emptyUncommittedAge0.add(new RegionStat(EMPTY_UNCOMMITTED, 0));
+            emptyCommittedAge3.add(new RegionStat(EMPTY_COMMITTED, 3));
+            trashAge6.add(new RegionStat(TRASH, 6));
+            humongousAge9.add(new RegionStat(HUMONGOUS, 9));
+            pinnedHumongousAge12.add(new RegionStat(PINNED_HUMONGOUS, 12));
+            cSetAge15.add(new RegionStat(CSET, 15));
+            pinned.add(new RegionStat(PINNED, 0));
+            pinnedCSet.add(new RegionStat(PINNED_CSET, 0));
+            tLab.add(new RegionStat(1.0f, 1.0f, 0.3f, 0.3f, 0.5f, 0.3f, YOUNG, REGULAR));
+            gcLab.add(new RegionStat(1.0f, 1.0f, 0.2f, 0.3f, 0.5f, 0.3f, YOUNG, REGULAR));
+            pLab.add(new RegionStat(1.0f, 1.0f, 0.7f, 0.8f, 0.9f, 0.5f, OLD, REGULAR));
+            sharedLabYoung.add(new RegionStat(1.0f, 1.0f, 0.2f, 0.3f, 0.5f, 0.7f, YOUNG, REGULAR));
+            sharedLabOld.add(new RegionStat(1.0f, 1.0f, 0.2f, 0.9f, 0.5f, 0.7f, OLD, REGULAR));
+        }
+        snapshot_1= new Snapshot(0, 1024, 1, emptyUncommittedAge0, 0, new Histogram(2));
```

Comment:
You could move each of these `snapshot_N` variables into their respective test case and just call it `snapshot`. They don't need to be member variables.

Reply:
Okay, my lastest commit should fix this part.